packages/ledgerstate/output.go

```diff
@@ -867,6 +895,977 @@ var _ Output = &SigLockedColoredOutput{}
+	o.id = outputID
+
+	return o
+}
+
+// Type returns the type of the Output which allows us to generically handle Outputs of different types.
+func (o *ExtendedLockedOutput) Type() OutputType {
+	return ExtendedLockedOutputType
+}
+
+// Balances returns the funds that are associated with the Output.
+func (o *ExtendedLockedOutput) Balances() *ColoredBalances {
+	return o.balances
+}
+
+// UnlockValid determines if the given Transaction and the corresponding UnlockBlock are allowed to spend the Output.
+func (o *ExtendedLockedOutput) UnlockValid(tx *Transaction, unlockBlock UnlockBlock, inputs []Output) (unlockValid bool, err error) {
+	if o.TimeLockedNow(tx.Essence().Timestamp()) {
+		return false, nil
+	}
+	addr := o.UnlockAddressNow(tx.Essence().Timestamp())
+
+	switch blk := unlockBlock.(type) {
+	case *SignatureUnlockBlock:
+		// unlocking by signature
+		unlockValid = blk.AddressSignatureValid(addr, tx.Essence().Bytes())
+
+	case *AliasUnlockBlock:
+		// unlocking by alias reference
+		refAliasOutput, isAlias := inputs[blk.AliasInputIndex()].(*AliasOutput)
+		if !isAlias {
+			return false, xerrors.New("ExtendedLockedOutput: referenced input must be AliasOutput")
+		}
+		if !addr.Equals(refAliasOutput.GetAliasAddress()) {
+			return false, xerrors.New("ExtendedLockedOutput: wrong alias referenced")
+		}
+		unlockValid = refAliasOutput.IsSelfGoverned() || !refAliasOutput.IsUnlockedForGovernanceUpdate(tx)
+
+	default:
+		err = xerrors.Errorf("ExtendedLockedOutput: unsupported unlock block type: %w", cerrors.ErrParseBytesFailed)
+	}
+	return
+}
+
+// Address returns the Address that the Output is associated to.
+func (o *ExtendedLockedOutput) Address() Address {
+	return o.address
+}
+
+// FallbackAddress returns the fallback address that the Output is associated to.
+func (o *ExtendedLockedOutput) FallbackAddress() (addy Address) {
+	if o.fallbackAddress == nil {
+		return
+	}
+	return o.fallbackAddress
+}
+
+// Input returns an Input that references the Output.
+func (o *ExtendedLockedOutput) Input() Input {
+	if o.ID() == EmptyOutputID {
+		panic("ExtendedLockedOutput: Outputs that haven't been assigned an ID, yet cannot be converted to an Input")
+	}
+
+	return NewUTXOInput(o.ID())
+}
+
+// Clone creates a copy of the Output.
+func (o *ExtendedLockedOutput) Clone() Output {
+	return &ExtendedLockedOutput{
+		id:       o.id,
+		balances: o.balances.Clone(),
+		address:  o.address.Clone(),
+	}
+}
+
+// UpdateMintingColor replaces the ColorMint in the balances of the Output with the hash of the OutputID. It returns a
+// copy of the original Output with the modified balances.
+func (o *ExtendedLockedOutput) UpdateMintingColor() Output {
+	coloredBalances := o.Balances().Map()
+	if mintedCoins, mintedCoinsExist := coloredBalances[ColorMint]; mintedCoinsExist {
+		delete(coloredBalances, ColorMint)
+		coloredBalances[Color(blake2b.Sum256(o.ID().Bytes()))] = mintedCoins
+	}
+	updatedOutput := NewExtendedLockedOutput(coloredBalances, o.Address())
```

Comment:
so if I create an extended output with fallback, timelock and metadata and I update the minting color, I only keep the address and balances but not the rest? That doesn't make sense for me.

Reply:
fixed. Was a copy from siglocked code